activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -759,29 +759,40 @@ def configure_connection
 
           super
 
+          pragmas = @config.fetch(:pragmas, {}).stringify_keys
+
           # Enforce foreign key constraints
           # https://www.sqlite.org/pragma.html#pragma_foreign_keys
           # https://www.sqlite.org/foreignkeys.html
-          raw_execute("PRAGMA foreign_keys = ON", "SCHEMA")
+          @raw_connection.foreign_keys = pragmas.fetch("foreign_keys", true)
           unless @memory_database
             # Journal mode WAL allows for greater concurrency (many readers + one writer)
             # https://www.sqlite.org/pragma.html#pragma_journal_mode
-            raw_execute("PRAGMA journal_mode = WAL", "SCHEMA")
+            @raw_connection.journal_mode = pragmas.fetch("journal_mode", :wal)
+            # raw_execute("PRAGMA journal_mode = #{pragmas.fetch("journal_mode", :wal)}", "SCHEMA")
             # Set more relaxed level of database durability
             # 2 = "FULL" (sync on every write), 1 = "NORMAL" (sync every 1000 written pages) and 0 = "NONE"
             # https://www.sqlite.org/pragma.html#pragma_synchronous
-            raw_execute("PRAGMA synchronous = NORMAL", "SCHEMA")
+            @raw_connection.synchronous = pragmas.fetch("synchronous", :normal)
             # Set the global memory map so all processes can share some data
             # https://www.sqlite.org/pragma.html#pragma_mmap_size
             # https://www.sqlite.org/mmap.html
-            raw_execute("PRAGMA mmap_size = #{128.megabytes}", "SCHEMA")
+            @raw_connection.mmap_size = pragmas.fetch("mmap_size", 128.megabytes)
           end
           # Impose a limit on the WAL file to prevent unlimited growth
           # https://www.sqlite.org/pragma.html#pragma_journal_size_limit
-          raw_execute("PRAGMA journal_size_limit = #{64.megabytes}", "SCHEMA")
+          @raw_connection.journal_size_limit = pragmas.fetch("journal_size_limit", 64.megabytes)
           # Set the local connection cache to 2000 pages
           # https://www.sqlite.org/pragma.html#pragma_cache_size
-          raw_execute("PRAGMA cache_size = 2000", "SCHEMA")
+          @raw_connection.cache_size = pragmas.fetch("cache_size", 2000)
+
+          pragmas.except("foreign_keys", "journal_mode", "synchronous", "mmap_size", "journal_size_limit", "cache_size").each do |pragma, value|
```

Comment:
Just a nit, but this code isn't DRY.   If I add an explicit set of a pragma above I need to remember to remove it here (well not really because it would be a noop, e.g. I'm just setting it twice to the same value)

Instead consider defining the defaults as a Hash and then merging them with the database configuration and then setting them all at once.

Example:

```
pragma_defaults = {
  'foreign_keys' => true,
  'synchronous' => :normal,
  'mmap_size' => 128.megabytes,
  'journal_size_limit' => 64.megabytes,
  'cache_size' => 2000
}

pragmas.merge(pragma_defaults).each do ....

```